### PR TITLE
Always build shared

### PR DESCRIPTION
**Pull request by mleotta**

### CODE REVIEW

**Comment by linus-sherrill** [comment]
+1

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,12 +45,8 @@ include( maptk-depends )
...
 include(CMakeDependentOption)
 
-# we can't build shared libraries on Windows so we leave it off by default
-if (WIN32)
-  option(BUILD_SHARED_LIBS "Build shared libraries." ON)
-else()
-  option(BUILD_SHARED_LIBS "Build shared libraries." OFF)
-endif()
+# Always build shared libraries, required due to plugin infrastructure
+set(BUILD_SHARED_LIBS ON)
```
Comment (Purg) [inline CMakeLists.txt]: I think this still lets a user turn it off?  Might want to force setting into cache instead of / in addition to this.

Comment (mleotta) [inline CMakeLists.txt]: I copied this approach from Vital.  It seems to work.  I don't see the BUILD_SHARED_LIBS option in the CMake UI (ccmake) or in the CMakeCache.txt file.  Is that not what you would expect?

Comment (Purg) [inline CMakeLists.txt]: Just tested this out with vital. A user can specify `-DBUILD_SHARED_LIBS:BOOL=OFF`, making a cache entry that says OFF, but the build proceeds to build shared.  Part of me finds this confusing, and that a warning or something should be displayed when the user tries to build statically stating that we don't allow it.

**Pull request merged**